service/debugger/debugger.go

```diff
@@ -420,6 +428,12 @@ func (d *Debugger) Restart(rerecord bool, pos string, resetArgs bool, newArgs []
 		p, err = d.recordingRun(run)
 		d.recordingDone()
 	} else {
+		if rebuild {
```

Comment:
This should go before the `if recorded` line, otherwise it won't have an effect on recorded targets.

Also, breakpoints that can't be set by (file, line) should be discarded in the `for _, oldBp := ...` loop below.